bevy_mod_scripting_core/src/systems.rs

```diff
@@ -95,11 +95,20 @@ pub fn script_remove_synchronizer<H: ScriptHost>(
     mut query: RemovedComponents<ScriptCollection<H::ScriptAsset>>,
     mut contexts: ResMut<ScriptContexts<H::ScriptContext>>,
 ) {
-    query.iter().for_each(|v| {
+    for v in query.iter() {
         // we know that this entity used to have a script component
         // ergo a script context must exist in ctxts, remove all scripts on the entity
-        contexts.remove_context(v.index());
-    })
+        let mut script_ids = Vec::new();
+        for (script_id, (entity, ..)) in contexts.context_entities.iter() {
+            if entity.index() == v.index() {
+                script_ids.push(*script_id);
+            }
+        }
+
+        for script_id in script_ids {
+            contexts.remove_context(script_id);
+        }
```

Comment:
```suggestion
        let script_ids = contexts.context_entities.iter().filter_map(|(script_id, (entity, ..))|  (entity == v).then_some(script_id));
        for script_id in  script_ids{
            contexts.remove_context();
        }
```

How about something like this?

Reply:
yep, that's better :)
Updated in the recent commit.